### PR TITLE
Fix async reboot to sync reboot

### PR DESCRIPTION
**Pull request by qingwangrh**
There just using sendline but not wait guest to shutdown.
It makes the connection reset.

ID:1816066,1816021,1777166
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=plug_cdrom --guestname=Win2019,RHEL.8.2 --clone=no
(1/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.plug_cdrom.with_hotplug.reboot_vm: PASS (127.61 s)
 (2/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (103.34 s)
 (3/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_hotplug.reboot_vm: PASS (228.73 s)
 (4/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (224.07 s)

**Comment by qingwangrh** [comment]
@YongxueHong @zhenyzha @peixiu Please help to review. thanks.

File: qemu/tests/plug_cdrom.py [hunk 1]
```diff
@@ -44,11 +44,8 @@ def _shutdown_vm(session):
 
     def _reboot_vm(session):
         """Reboot vm."""
-        reboot_cmd = params["reboot_cmd"]
-        error_context.context(
-            "Rebooting VM by \"%s\"." % reboot_cmd, logging.info)
-        session.sendline(reboot_cmd)
-        return vm.wait_for_login(timeout=360)
+        error_context.context("Rebooting VM.", logging.info)
+        return vm.reboot(session=session, timeout=360)
```
Comment (YongxueHong) [inline qemu/tests/plug_cdrom.py]: The `vm.reboot` will send comand `reboot_cmd = self.params.get("reboot_command")` which is `reboot_command = shutdown -r now` in linux, but this case sends the command is "reboot".
```
    variants:
        - reboot_vm:
            Windows:
                reboot_cmd = shutdown -t 0 -r
                shutdown_cmd = shutdown -t 0 -s
            Linux:
                reboot_cmd = reboot
                shutdown_cmd = shutdown -h now
```
Maybe you could update the manul case steps if it doesn't affect your testing.
Or change the `reboot_command` for this case.
CC @peixiu

Comment (qingwangrh) [reply qemu/tests/plug_cdrom.py]: Cases updated.

**Comment by qingwangrh** [comment]
This issue has gone after upgrade lastest avocado-vt,so close it

python ConfigTest.py --testcase=plug_cdrom.with_hotplug.reboot_vm.i440fx --guestname=RHEL.8.2.0 --clone=no --customsparams="qemu_force_use_drive_expression = yes" --nrepeat=5

(1/5) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.plug_cdrom.with_hotplug.reboot_vm: PASS (51.90 s)
 (2/5) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.plug_cdrom.with_hotplug.reboot_vm: PASS (48.61 s)
 (3/5) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.plug_cdrom.with_hotplug.reboot_vm: PASS (48.57 s)
 (4/5) repeat4.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.plug_cdrom.with_hotplug.reboot_vm: PASS (48.57 s)
 (5/5) repeat5.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.plug_cdrom.with_hotplug.reboot_vm: PASS (49.19 s)

**Comment by qingwangrh** [comment]
But there still have issue on plug_cdrom.with_unplug.reboot_vm for windows
python ConfigTest.py --testcase=plug_cdrom.with_unplug.reboot_vm --guestname=Win2019 --clone=no --customsparams="qemu_force_use_drive_expression = yes" --nrepeat=5

(1/5) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: FAIL: Failed to shutdown vm. (125.68 s)
 (2/5) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (75.90 s)
 (3/5) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (75.73 s)
 (4/5) repeat4.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (74.24 s)
 (5/5) repeat5.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (75.63 s)

The reason is have issue in previous implemenation of _reboot_vm, 
after apply this patch, it can run well.
python ConfigTest.py --testcase=plug_cdrom.with_unplug.reboot_vm --guestname=Win2019 --clone=no --customsparams="qemu_force_use_drive_expression = yes" --nrepeat=10

(01/10) repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (102.29 s)
 (02/10) repeat2.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (75.73 s)
 (03/10) repeat3.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (76.47 s)
 (04/10) repeat4.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (79.89 s)
 (05/10) repeat5.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (75.62 s)
 (06/10) repeat6.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (75.10 s)
 (07/10) repeat7.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (77.00 s)
 (08/10) repeat8.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (75.90 s)
 (09/10) repeat9.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (81.19 s)
 (10/10) repeat10.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.plug_cdrom.with_unplug.reboot_vm: PASS (80.53 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

File: qemu/tests/plug_cdrom.py [hunk 2]
```diff
@@ -109,4 +106,5 @@ def _check_cdrom_info(items):
         if action == HOTPLUG:
             _check_cdrom_info(items_checked)
 
-    _shutdown_vm(_reboot_vm(session))
+    session = _reboot_vm(session)
+    _shutdown_vm(session)
```
Comment (YongxueHong) [inline qemu/tests/plug_cdrom.py]: Why separate it?

Comment (qingwangrh) [reply qemu/tests/plug_cdrom.py]: updated

**Pull request merged**